### ISSUE TITLE
RandAugment implementation differences.

### ISSUE THREAD

**Issue by sebastian-sz** [keras-team/keras-cv#411]
I'm reading `keras_cv` RandAugment implementation and feel like there are some differences with [TF implementation](https://github.com/tensorflow/tpu/blob/8aac591077e5781785aa6c22bc400472ba14dada/models/official/efficientnet/autoaugment.py#L672) referenced in the paper. I'm still reading this layer so I might not have understood everything, but here it goes:

* Missing `Posterization` op in RandAugment - [reference TF Implementation](https://github.com/tensorflow/tpu/blob/8aac591077e5781785aa6c22bc400472ba14dada/models/official/efficientnet/autoaugment.py#L694).
* Missing  `Sharpness` op in RandAugment - this op is present as a [Random Layer](https://github.com/keras-team/keras-cv/blob/e2a8874e8b875de1cc8b335961bf766da93cadf9/keras_cv/layers/preprocessing/random_sharpness.py#L20) in keras-cv.
* Missing `Rotate` op in RandAugment - discussed also in #402 .
* Missing random magnitude negation in `shear` and `translate` ops - reference TF implementation [randomly negates](https://github.com/tensorflow/tpu/blob/8aac591077e5781785aa6c22bc400472ba14dada/models/official/efficientnet/autoaugment.py#L470) magnitudes for these ops, and it looks like it's also [used in RandAugment](https://github.com/tensorflow/tpu/blob/8aac591077e5781785aa6c22bc400472ba14dada/models/official/efficientnet/autoaugment.py#L705).
* Different behaviour for `Random*` layers, e.g `RandomBrightness` is used instead of `Brightness`, but the function behaves differently in [keras](https://github.com/keras-team/keras/blob/e6b80d49573758b16f0c05187022006062896e03/keras/layers/preprocessing/image_preprocessing.py#L1571) and differently in [TF implementation](https://github.com/tensorflow/tpu/blob/8aac591077e5781785aa6c22bc400472ba14dada/models/official/efficientnet/autoaugment.py#L216). Perhaps this is marginal. 

Are those differences intentional / are the missing operations not so important or should they be added in the future?

**Comment by bhack** [comment]
I don't know how many points was covered in private threads https://github.com/keras-team/keras-cv/issues/84

**Comment by sebastian-sz** [comment]
@bhack If those differences were discussed and accepted maybe they could be described in the docstring. 

For example: the `Identity` operation is also missing but it's mentioned that this is substituted by the `rate` parameter.

**Comment by bhack** [comment]
I've correctly updated the wrong mentioned ticket

**Comment by sebastian-sz** [comment]
I see, #169 is closed but there is no mention whether it was decided to ignore or include `Posterize`. I still believe that a docstring mention would be good to have for such cases.

Also I can't find information/discussion regarding presence of other ops: `Sharpness` & `Rotate`.

**Comment by bhack** [comment]
Yes, I think also that some underground information/choices could also be linked to this kind of arguments https://github.com/keras-team/keras-cv/pull/294#discussion_r848820525

**Comment by LukeWood** [comment]
I actually discussed this heavily with Ekin and am confident that our implementation is correct and that the others are actually *NOT* correct!

I can take an action item to discuss this in a "why KerasCV doc"; I am actively working with several original authors to verify our implementations of various components.

**Comment by sebastian-sz** [comment]
@LukeWood thank you for the answer. 

I did not know about internal work or discussions and was simply worried that there are quite few differences between `keras_cv` and other existing implementations in various libraries. If they were discussed and even approved by original authors then it sounds solid to me.

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new **KerasHub** package, which supports image, text, and multi-modal models. Please read [the announcement](https://github.com/keras-team/keras-hub/issues/1831). `KerasHub` will support all the core functionality of `KerasCV`.

`KerasHub` can be installed with `!pip install -U keras-hub`. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
